daemon/container_operations.go

```diff
@@ -339,7 +339,7 @@ func (daemon *Daemon) updateContainerNetworkSettings(container *container.Contai
 	)
 
 	mode := container.HostConfig.NetworkMode
-	if container.Config.NetworkDisabled || mode.IsContainer() {
+	if container.Config.NetworkDisabled || mode.IsContainer() || mode.IsNetNs() {
```

Comment:
I cannot test this patch with 1.12.6 because I'm on Fedora 26 but I ported the patch to 1.13.1 and I noticed there are other code paths in this file where you should call `mode.IsNetNs()` and exit before docker can continue setting up network.